src/OpenTelemetry.Extensions.Enrichment/EnrichmentExtensions.cs

```diff
@@ -0,0 +1,121 @@
+// <copyright file="EnrichmentExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry.Extensions.Enrichment;
+
+public static class EnrichmentExtensions
+{
+    public static TracerProviderBuilder AddTraceEnricher<T>(this TracerProviderBuilder builder)
+        where T : TraceEnricher
+    {
+        Guard.ThrowIfNull(builder);
+
+        return builder
```

Comment:
Help me to see if the following understanding is correct:
1. TryAddEnrichment adds TraceEnrichmentProcessor to DI.
2. The configure callback adds the enricher to DI. (so that DI will have a list of enrichers)
3. TraceEnrichmentProcessor asks for IEnumerable<TraceEnrichers> from DI (ctor), and stores the list.
4. TraceEnrichmentProcessor calls enrich on all the above.

Reply:
Yes, that's right for the DI scenario.

Additionally, users can add enrichers via such methods as `public static TracerProviderBuilder AddTraceEnricher(this TracerProviderBuilder builder, TraceEnricher enricher`) which do not require DI, so internally it will also be added to the list of enrichers in the processor